client/blocks/get-apps/mobile-download-card.tsx

```diff
@@ -17,12 +15,12 @@ const MobileDownloadCardTest = () => {
 			title={ translate( 'Jetpack mobile app for WordPress' ) }
 			subtitle={ translate( 'Create, design, manage, and grow your WordPress website.' ) }
 		>
-			{ isMobile ? (
+			{ isMobile() ? (
 				<div className="get-apps__badges">
 					<AppsBadge
-						storeName={ isIos ? 'ios' : 'android' }
+						storeName={ isIos() ? 'ios' : 'android' }
 						utm_source="calypso"
-						utm_campaign={ isIos ? 'calypso-get-apps-button' : 'calypso-get-apps' }
+						utm_campaign={ isIos() ? 'calypso-get-apps-button' : 'calypso-get-apps' }
```

Comment:
The `AppPromo` component has a very similar code to distinguish between iOS and Android, but although in both components it does the same thing, they are not identical.
- `AppPromo` detects `isIos()` and `isAndroid()` separately, which is the right way
- here `isAndroid` is calculated as `isMobile() && ! isIos()` which is wrong: it classifies the long tail of other mobile OSes as Android, but they are not Android

In both components we show links to the two major app stores, so all the other mobile OSes should be ignored.

Reply:
> The `AppPromo` component has a very similar code to distinguish between iOS and Android, but although in both components it does the same thing, they are not identical.
> 
> * `AppPromo` detects `isIos()` and `isAndroid()` separately, which is the right way
> * here `isAndroid` is calculated as `isMobile() && ! isIos()` which is wrong: it classifies the long tail of other mobile OSes as Android, but they are not Android
> 
> In both components we show links to the two major app stores, so all the other mobile OSes should be ignored.

Yeah these are good points. I updated to check specifically for iOS and Android in e51ddc5d10b (test coverage added in d6ff89832c0).